### PR TITLE
4434 - Readme file updated for react set up

### PR DESCRIPTION
**Pull request by shrutichatti**
Readme.md file has been updated for froala editor set up in react framework

### CODE REVIEW

**Comment by shrutichatti** [comment]
Hi @harasunu-narayan 

I have updated the Read me file with proper separation between usage for class component and function component

The file contains two section **usage with class component** and **usage with functional component**
1) the new added part functional component contains the syntax changes how the froala would be integrated wrt to latest react version, however there are no logical changes.

2) Content which common to both for example Custom button has been kept after the Usage with functional component section

**Comment by shrutichatti** [comment]
hi @harasunu-narayan 

I have handled the issue of react strict mode render twice , for this have maintained one flag `createEditorFlag` to maintain the count of number of times the editor get initialized in  `createEditor` function. 

As per the behavior of the react strictmode in react 18 which trigger the double render , `createEditor` function get called twice as a result the double render is happening.

To handle this maintained the count which make sure the editor get initialized only once when the `createEditorFlag==0`

**Comment by shrutichatti** [comment]
Hi @harasunu-narayan 

After thorough testing , i have handled the render twice issue by keeping track of the id that is created during each initialization. As the per the behavior of the react strict mode render happens twice so keep track of the instance id is used . 
```
 if(this.editor && this.editor.id != this.editor.sid && createEditorFlag!=this.editor.id){
      this.editor = new FroalaEditor(this.element, this.config);
    }else if(!this.editor){
      this.editor = new FroalaEditor(this.element, this.config);
    }
```
 whenever the editor function is called there is id and sid that is generated from the core froala , which can be used to track the instance of the editor . Also `createEditorFlag` is used to keep the track of the last instance of editor .
                for example whenever the instance is created ids genearated are ` id = 1 and sid=1` , second instance id=2 and sid=1 
Hence the first if condition make sure that the repeated intiliazation for the same instance is not created 

The else case is used to handle  first instance editor which is always null 


Have tested by creating the multiple instance of the editor , initializing the editor on the image, a ,button,input tag . Every time the editor is initialized only once .

**Review by harasunu-narayan — COMMENTED** [review]
@shrutichatti ,
Provided solution is breaking the manual initialization/destroy implementation. Refer recording https://recordit.co/J2ij4cA0x9

**Comment by shrutichatti** [comment]
hi @harasunu-narayan 

I have fixed the bug related to manual initialization , also there were bug in the other  demo example from beginning have fixed them as well . Kindly review the same

**Comment by shrutichatti** [comment]
Hi @harasunu-narayan 

I have removed the extra condition added and the create editor flag and tested across these following scenario
1) SDK all the demo example are working properly 
2) React 18 sample app having one page with multiple editor instance working properly
3) React 18 sample app having two pages and multiple editor instance working properly 

please check the below recording for the 3 scenario 
https://recordit.co/XJMbzqOsx1

Kindly review the same

File: lib/FroalaEditorFunctionality.jsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ export default class FroalaEditorFunctionality extends React.Component {
   }
 
   componentWillUnmount() {
+    this.element=null
```
Comment (harasunu-narayan) [inline lib/FroalaEditorFunctionality.jsx:51]: Why is this required? This will cause the `destroyEditor` method to fail.

Comment (shrutichatti) [reply lib/FroalaEditorFunctionality.jsx]: Hi @harasunu-narayan 

I have reverted back this line of code as it was causing `destroyEditor` method to fail

**Comment by shrutichatti** [comment]
Hi @harasunu-narayan 

I have found that for every instance when inside ` destroyEditor() ` method ` this.editor.destroy();` method of core froala is called the value of  ` this.destrying = true ` is set inside the core froala.

Now what is happening with the multiple render is that suppose  for one textarea we have two render that happens as per the strict mode behavior so during the first destroy method call the value of this.destrying is set to true for that instance as a result when the second render happen the destroy method is not executed for the same instance. 

Hence in the `destroyEditor()` we will have to set value of this.destrying so that it can successfully handle the double render scenario of react 18 strict mode

File: lib/FroalaEditorFunctionality.jsx [hunk 2]
```diff
@@ -200,6 +200,19 @@ export default class FroalaEditorFunctionality extends React.Component {
       this.listeningEvents.length = 0;
       this.element = null;
       this.editorInitialized = false;
+      this.config = {
+        immediateReactModelUpdate: false,
+        reactIgnoreAttrs: null
+      };
+      let tagName = this.el.tagName.toLowerCase();
+      if (this.SPECIAL_TAGS.indexOf(tagName) == -1) {
+        if(this.editor && this.editor.destrying && !this.props.onManualControllerReady && this.tag=="textarea"){
```
Comment (harasunu-narayan) [inline lib/FroalaEditorFunctionality.jsx:209]:
```suggestion
        if(this.editor && this.editor.destrying && !this.props.onManualControllerReady && this.tag == 'textarea'){
```

**Review by harasunu-narayan — CHANGES_REQUESTED** [review]

**Comment by shrutichatti** [comment]
Hi @harasunu-narayan 
I have done changes suggested kindly verify

**Review by harasunu-narayan — APPROVED** [review]
